cmd/snap/cmd_connections.go

```diff
@@ -0,0 +1,222 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+
+	"github.com/jessevdk/go-flags"
+)
+
+type cmdConnections struct {
+	clientMixin
+	All          bool `long:"all"`
+	Disconnected bool `long:"disconnected"`
+	Positionals  struct {
+		Snap installedSnapName `skip-help:"true"`
+	} `positional-args:"true"`
+}
+
+var shortConnectionsHelp = i18n.G("List interface connections")
+var longConnectionsHelp = i18n.G(`
+List connections between plugs and slots. When passed an optional
+snap name, lists connected and unconnected plugs and slots for
+that snap only.
+
+Pass --all to list connected and unconnected plugs and slots.
+`)
```

Comment:
long helps should start with "The *foo* command ...".

Reply:
Tried to tweak the help information a little, but it's likely I came up with something even worse in the process. Would appreciate your and @degville's input on this.

Reply:
I think it's better to be consistent with our other long help output (eg. aliases or interfaces), which would make it similar to the following:

```
The connections command lists connections between plugs and slots
in the system. Pass --all to list unconnected plugs and slots 
alongside those already connected.

$ snap connections <snap>

Lists connected and unconnected plugs and slots for the specified
snap.                                                         
```